### PR TITLE
Popup not opening

### PR DESCRIPTION
**Pull request by jhonatasrm**
### Requirement for Contributing a Pull Request

* [X] Put an X between the brackets on this line if you have done all of the 
      following:
    * You agree to license your code under the project's open source license 
      (MPL 2.0).
    * Your branch is based off the current master.
    * You did not include merge commits in pull requests; include only commits 
      with the new relevant code.

### Identify the Bug

#206 

### Description of the Change

Was added try-catch when verifying the chosen product.

### Alternate Designs

no

### Possible Drawbacks

no

### CODE REVIEW

File: src/js/background.js
Comment (dannycolin) [inline src/js/background.js:271]: We should leave the `catch` empty to avoid debugging logs in the production version.

**Comment by dannycolin** [comment]
@jhonatasrm Do you have step-to-reproduce instructions? I'm concerned that simply wrapping it in a try...catch results in silently failing to remove firefox-preview from the localStorage. We should probably find a solution that always successed to remove it.

**Review by WesleyBranton — CHANGES_REQUESTED** [review]
I would prefer a more generic approach, since it seems strange to have a rule exclusively for firefox-preview. Instead, maybe we should have a constant array of all valid products and then filter out any from the storage API that do not appear in the list.

File: src/js/background.js
Comment (WesleyBranton) [reply src/js/background.js:271]: I'm torn on this. I'd much rather have it logged as a proper console.error(). I feel that using an empty catch to silently suppress an error is not good practice.

**Comment by jhonatasrm** [comment]
> I would prefer a more generic approach, since it seems strange to have a rule exclusively for firefox-preview. Instead, maybe we should have a constant array of all valid products and then filter out any from the storage API that do not appear in the list.

I agree.. I don't know if we need that rule tbh, once he was removed right?

**Comment by jhonatasrm** [comment]
> @jhonatasrm Do you have step-to-reproduce instructions? 

Yes, in the issue #206.

**Comment by jhonatasrm** [comment]
The verification for 'firefox-preview' has been removed, as discussed earlier. For some reason that part was checking if the chooseProduct contained the 'firefox-preview'.

File: src/js/background.js
Comment (jhonatasrm) [reply src/js/background.js]: The console.error instead of the console.log, was added to be more visible on the console, and also the part that verifies the 'firefox-preview' was removed.

**Comment by jhonatasrm** [comment]
@WesleyBranton and @dannycolin, do you see any further improvements we should make before merging this PR and publishing the new version with the NL language?

**Comment by dannycolin** [comment]
I don't see anything on my side.

**Comment by WesleyBranton** [comment]
Same here

**Comment by jhonatasrm** [comment]
@dannycolin  and @WesleyBranton, I removed the verification for 'firefox-preview'  as well, once we dont need more that. Apparently is working fine, if possible make the review and test.

Thank you very much.

**Review by dannycolin — APPROVED** [review]

**Review by dannycolin — COMMENTED** [review]
This fix the popup. However, if a user still has `firefox-preview` in `data.chooseProduct` in the extension storage, it prevents the option page to open.

@jhonatasrm can you add a function that looks if `firefox-desktop` exists in the storage and removes it if it does?

**Comment by jhonatasrm** [comment]
@dannycolin , I've now implemented a check for the presence of 'firefox-preview' in the extension storage, and if found, it is automatically removed.

image

**Comment by dannycolin** [comment]
LGTM.

**Comment by dannycolin** [comment]
> I would prefer a more generic approach, since it seems strange to have a rule exclusively for firefox-preview. Instead, maybe we should have a constant array of all valid products and then filter out any from the storage API that do not appear in the list.

@WesleyBranton I agree but I think this can by implemented in a follow-up bug. Do you mind filing one? 

@jhonatasrm I approved. I'll let you the honour to press the merge button. Do you need any help to push the new release to AMO?

**Comment by jhonatasrm** [comment]
> @jhonatasrm I approved. I'll let you the honour to press the merge button. Do you need any help to push the new release to AMO?

As for the next steps, we could either create a new branch and PR for the version update and README changes, or just push the updates directly on this branch?  About the new release in AMO, I think is fine.

**Comment by dannycolin** [comment]
IIRC, we're using tags. So the steps are something like:

1. Merge this PR in main
2. Create a new PR that changes the version in manifest.json
3. Tag this commit as 1.6
4. Build a xpi with your branch positioned to the tag you created
5. Upload the build to AMO

**Pull request merged**
